### PR TITLE
Add what's new for time performance improvements

### PR DESCRIPTION
**Pull request by taldcroft**
### Description

This pull request is to add what's new (4.2) for time performance improvements.

### CODE REVIEW

File: docs/time/index.rst [hunk 1]
```diff
@@ -705,6 +705,9 @@ The current time can be determined as a |Time| object using the
 
 The two should be very close to each other.
 
+
+.. _time-fast-c-parser
```
Comment (pllim) [inline docs/time/index.rst:709]:
```suggestion
.. _time-fast-c-parser:
```

File: docs/whatsnew/4.2.rst [hunk 2]
```diff
@@ -35,9 +35,23 @@ and has become the default cosmology.  It is identical to the previous, prelimin
 version (Planck18_arXiv_v2), which is deprecated and will be removed in a future release.
 
   >>> from astropy.cosmology import Planck18
-  >>> Planck18.age(0)  
+  >>> Planck18.age(0)
   <Quantity 13.7868853 Gyr>
 
+
```
Comment (pllim) [inline docs/whatsnew/4.2.rst:41]: Need a ref link for the section and then link to this section from TOC above.

File: docs/whatsnew/4.2.rst [hunk 3]
```diff
@@ -35,9 +35,23 @@ and has become the default cosmology.  It is identical to the previous, prelimin
...
+The performance for creating a ``Time`` object from a large array of
+fixed-format string times was dramatically improved. For ISO and ISOT formats
+the speed-up is a factor of 25 and for the year day-of-year format the speedup
+is a factor of 50. This is done with a new C-based time string parser which can
+also be used for custom user-defined Time formats. For details see
+:ref:`time-fast-c-parser`.
+
+In addition the performance for creating a scalar ``Time`` object in a epoch
+format like ``unix``, ``unix_tai`` or ``cxcsec`` was improved by a factor of 4.
```
Comment (pllim) [inline docs/whatsnew/4.2.rst:53]: Oxford comma! 😹 

```suggestion
format like ``unix``, ``unix_tai``, or ``cxcsec`` was improved by a factor of 4.
```

**Review by pllim — CHANGES_REQUESTED** [review]
Thanks! Need some minor fixes.

I also wonder if we have benchmark that would show such improvements visually over at https://github.com/astropy/astropy-benchmarks/ . More of a personal curiosity, as we don't have a precedence of link to benchmarks from What's New.

**Review by pllim — APPROVED** [review]
To speed things up, I applied my review as follow-up commits, so LGTM now. Thanks!

**Comment by taldcroft** [comment]
@pllim - thanks for the cleanup, and sorry for not getting it right the first time. I don't get along with sphinx... and I always seem to forget that Oxford comma. :smile:

**Pull request merged**